dbt/adapters/sql/connections.py

```diff
@@ -199,3 +211,46 @@ def commit(self):
         connection.transaction_open = False
 
         return connection
+
+    def _retryable_cursor_execute(self,
+                                  execute_fn: Callable,
+                                  sql: str,
+                                  bindings: Optional[Any] = None,
+                                  retryable_exceptions: Iterable[Type[Exception]] = [],
+                                  retry_limit: int = 1,
+                                  retry_timeout: Union[Callable[[int], SleepTime], SleepTime] = 1,
+                                  _attempts: int = 0,
+                                  ) -> None:
+        timeout = retry_timeout(_attempts) if callable(retry_timeout) else retry_timeout
+        if timeout < 0:
+            raise DbtRuntimeError(
+                "retry_timeout cannot be negative or return a negative time."
+            )
+
+        try:
+            execute_fn(sql, bindings)
+
+        except tuple(retryable_exceptions) as e:
```

Comment:
This also feels like early abstraction. Can we update the type to be `Tuple` from start since the user of this is an adapter maintainer and not an end user?